### ISSUE TITLE
Error __DEFINE_ENABLE_IF_ 2

### ISSUE THREAD

**Issue by AnantaYudica** [AnantaYudica/basic#8]
```
#include "macro/EnableIf.h"

template<typename T>
struct A
{};

#define __DEFINE_ENABLE_IF_TMPL_PARAM_LIST_PRIMARY_TPLP1(\
    __T_NAME__,\
    __T_OUT_NAME__)\
typename __T_NAME__, template<typename> class __T_OUT_NAME__

#define __DEFINE_ENABLE_IF_TMPL_PARAM_LIST_SPECIALIZED_TPLS1(\
    __T_NAME__,\
    __T_OUT_NAME__)\
typename __T_NAME__, template<typename> class  __T_OUT_NAME__

#define __DEFINE_ENABLE_IF_ALIAS_TYPE_TMPL_ARG_LIST_ATTAL1()\
T1

/** 
 *  template<bool BoolTest, typename T1, template<typename> class To>
 *  struct EnableIf4
 *  {
 *      using type = To<T1>;
 *      using Type = To<T1>;
 *  };
 *  template<typename T1, template<typename> class To>
 *  struct EnableIf4<true, T1, To, false>
 *  {};
 *  template<typename T1, template<typename> class To>
 *  struct EnableIf4<false, T1, To, true>
 *  {};
 *  template<typename T1, template<typename> class To>
 *  struct EnableIf4<false, T1, To, false>
 *  {};
 * */
__DEFINE_ENABLE_IF_(EnableIf4, BoolTest, T1, To, TPLP1, TPLS1,, 
    ALIAS_TYPE_TMPL_,, ATTAL1, true);

int main()
{
   
}
```
`Error : '__DEFINE_ENABLE_IF_ALIAS_TYPE_TMPL_PARAM_LIST_' has not been declared`

**Comment by AnantaYudica** [comment]
Please Update parameters __DEFINE_ENABLE_IF_ before test 
```
__DEFINE_ENABLE_IF_(\
    __STRUCT_NAME__,\
    __BOOL_TEST_NAME__,\
    __T_NAME__,\
    __T_OUT_NAME__,\
    __TMPL_PARAM_LIST_PRIMARY_POSTFIX__,\
    __TMPL_PARAM_LIST_SPECIALIZED_POSTFIX__,\
    __TMPL_ARG_LIST_POSTFIX__,\
    __T_OUT_NAME_OR_T_OUT_NAME_TMPL__,\
    __T_OUT_NAME_TMPL_ARG_LIST_POSTFIX__,\
    __ALIAS_TYPE_OR_ALIAS_TYPE_TMPL,\
    __ALIAS_TYPE_TMPL_PARAM_LIST_POSTFIX__,\
    __CONDITIONAL__, ...)
```
Fixed in v0.1-beta-2

**Comment by AnantaYudica** [comment]
Update **from** :
```
#define __DEFINE_ENABLE_IF_ALIAS_TYPE_TMPL_ARG_LIST_ATTAL1()\
T1
```

```
__DEFINE_ENABLE_IF_(EnableIf4, BoolTest, T1, To, TPLP1, TPLS1,, 
    ALIAS_TYPE_TMPL_,, ATTAL1, true);
```
**to** :
```
#define __DEFINE_ENABLE_IF_T_OUT_NAME_TMPL_ARG_LIST_TONTAL1()\
T1
```
```
__DEFINE_ENABLE_IF_(EnableIf4, BoolTest, T1, To, TPLP1, TPLS1,, 
    T_OUT_NAME_TMPL_, TONTAL1, ALIAS_TYPE_,, true);
```
